src/errors.rs

```diff
@@ -75,28 +75,25 @@ pub enum ContextualError {
     /// Might occur when trying to access a page that does not exist
     #[error("Route {0} could not be found")]
     RouteNotFoundError(String),
-
-    /// In case miniserve was invoked without an interactive terminal and without an explicit path
-    #[error("Refusing to start as no explicit serve path was set and no interactive terminal was attached
-Please set an explicit serve path like: `miniserve /my/path`")]
-    NoExplicitPathAndNoTerminal,
-
-    /// In case miniserve was invoked with --no-symlinks but the serve path is a symlink
-    #[error("The -P|--no-symlinks option was provided but the serve path '{0}' is a symlink")]
-    NoSymlinksOptionWithSymlinkServePath(String),
 }
 
-impl ResponseError for ContextualError {
+impl ResponseError for RuntimeError {
     fn status_code(&self) -> StatusCode {
+        use RuntimeError as E;
+        use StatusCode as S;
         match self {
-            Self::ArchiveCreationError(_, err) => err.status_code(),
-            Self::RouteNotFoundError(_) => StatusCode::NOT_FOUND,
-            Self::InsufficientPermissionsError(_) => StatusCode::FORBIDDEN,
-            Self::InvalidHttpCredentials => StatusCode::UNAUTHORIZED,
-            Self::InvalidHttpRequestError(_) => StatusCode::BAD_REQUEST,
-            Self::DuplicateFileError => StatusCode::FORBIDDEN,
-            Self::UploadForbiddenError => StatusCode::FORBIDDEN,
-            _ => StatusCode::INTERNAL_SERVER_ERROR,
+            E::IoError(_, _) => S::INTERNAL_SERVER_ERROR,
+            E::MultipartError(_) => S::BAD_REQUEST,
+            E::DuplicateFileError => S::CONFLICT,
+            E::UploadForbiddenError => S::FORBIDDEN,
+            E::InvalidPathError(_) => S::BAD_REQUEST,
+            E::InsufficientPermissionsError(_) => S::FORBIDDEN,
+            E::ParseError(_, _) => S::BAD_REQUEST,
+            E::ArchiveCreationError(_, err) => err.status_code(),
+            E::ArchiveCreationDetailError(_) => S::INTERNAL_SERVER_ERROR,
+            E::InvalidHttpCredentials => S::UNAUTHORIZED,
+            E::InvalidHttpRequestError(_) => S::BAD_REQUEST,
+            E::RouteNotFoundError(_) => S::NOT_FOUND,
```

Comment:
Oh, I really like this! I didn't know we were hiding so many errors!

Reply:
Yeah this is the one thing I ***REALLY*** dislike about non-specific wildcard match clauses. When you add a new enum variant, there's no indication from the LSP that you probably also want to update the match block. I'd say it's almost as bad as [`enum_glob_use`](https://rust-lang.github.io/rust-clippy/master/#/enum_glob_use).